components/data-bridge/org.wso2.carbon.databridge.receiver.binary/src/main/java/org/wso2/carbon/databridge/receiver/binary/internal/BinaryDataReceiver.java

```diff
@@ -153,97 +154,88 @@ private void startSecureTransmission() throws IOException, DataBridgeException {
             context.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
 
             SSLServerSocketFactory sslServerSocketFactory = context.getServerSocketFactory();
-            Thread thread = new Thread(new BinarySecureEventServer(sslServerSocketFactory));
+            SSLServerSocket sslserversocket =
+                    (SSLServerSocket) sslServerSocketFactory.
+                            createServerSocket(binaryDataReceiverConfiguration.getSSLPort());
+            String sslProtocols = binaryDataReceiverConfiguration.getSslProtocols();
+            if (sslProtocols != null && sslProtocols.length() != 0) {
+                String[] sslProtocolsArray = sslProtocols.split(",");
+                sslserversocket.setEnabledProtocols(sslProtocolsArray);
+            }
+            String ciphers = binaryDataReceiverConfiguration.getCiphers();
+            if (ciphers != null && ciphers.length() != 0) {
+                String[] ciphersArray = ciphers.split(",");
+                sslserversocket.setEnabledCipherSuites(ciphersArray);
+            } else {
+                sslserversocket.setEnabledCipherSuites(sslserversocket.getSupportedCipherSuites());
+            }
+            Thread thread = new Thread(new BinarySecureEventServerAcceptor(sslserversocket));
             thread.start();
-
+            log.info("Started Binary SSL Transport on port : " + binaryDataReceiverConfiguration.getSSLPort());
         } catch (Exception e) {
             throw new DataBridgeException("Error when starting binary agent server ", e);
         }
     }
 
-    private void startEventTransmission() {
-        ServerSocketFactory serverSocketFactory = ServerSocketFactory.getDefault();
-        Thread thread = new Thread(new BinaryEventServer(serverSocketFactory));
+    private void startEventTransmission() throws IOException {
+        ServerSocketFactory serversocketfactory = ServerSocketFactory.getDefault();
+        ServerSocket serversocket = serversocketfactory.createServerSocket(binaryDataReceiverConfiguration.getTCPPort());
+        Thread thread = new Thread(new BinaryEventServerAcceptor(serversocket));
         thread.start();
+        log.info("Started Binary TCP Transport on port : " + binaryDataReceiverConfiguration.getTCPPort());
+    }
+
+    @Override
+    public void completingServerStartup() {
+        try {
+            start();
+        } catch (IOException e) {
+            log.error("Error while starting binary data receiver ", e);
```

Comment:
We can add a more specific error msg for the 2 catch blocks.

Reply:
Line 193 can be changed with a specific message. 
But shall we keep 195 as it is since **startSecureTransmission** function is throwing this DataBridgeException in several places and the messages will be included in e? Keeping the generic message here (195) seems to be okay since we cannot specify the exact message.

Reply:
Fixed the line 193 via https://github.com/wso2/carbon-analytics-common/pull/760/commits/a95b3ba138cf100657bebc4dc369cfc2862a6b02